File: app/src/main/java/com/estasvegano/android/estasvegano/data/web/interceptor/LocalizationInterceptor.java
package com.estasvegano.android.estasvegano.data.web.interceptor;

import android.support.annotation.NonNull;

import java.io.IOException;
import java.util.Locale;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;


public class LocalizationInterceptor implements Interceptor
{
    @Override
    @NonNull
    public Response intercept(@NonNull Chain chain) throws IOException
    {
        Request request = chain.request();
        request = request.newBuilder()
                .addHeader("Accept-Language", Locale.getDefault().getCountry())
                .build();

        Response response = chain.proceed(request);

        return response;
    }
}

File: app/src/main/java/com/estasvegano/android/estasvegano/data/web/interceptor/LocalizationInterceptor.kt
package com.estasvegano.android.estasvegano.data.web.interceptor

import okhttp3.Interceptor
import okhttp3.Response
import java.util.*

class LocalizationInterceptor : Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()
        request = request.newBuilder()
                .addHeader("Accept-Language", Locale.getDefault().country)
                .build()

        return chain.proceed(request)
    }
}
